edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Serializer/MetricsSerializer.cs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Drawing;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+
+    /// <summary>
+    /// Provides a way to serialize the datetime and value component of a metric.
+    /// </summary>
+    public class RawMetricValue
+    {
+        public const int Size = sizeof(long) + sizeof(double);
+        public DateTime TimeGeneratedUtc { get; set; }
+        public double Value { get; set; }
+
+        public static IEnumerable<byte> RawValuesToBytes(IEnumerable<RawMetricValue> rawValues)
```

Comment:
1. implement constructor
2. remove RawValuesToBytes method; and create a method "ToBytes" which just has the logic "BitConverter.GetBytes(t.TimeGeneratedUtc.Ticks).Concat(BitConverter.GetBytes(t.Value))"; then you can use this method from caller.

Reply:
I wanted to keep the methods symmetric, so RawValuesToBytes matches BytesToRawValues